AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/AWSDataStorePlugin.swift

```diff
@@ -98,6 +102,9 @@ final public class AWSDataStorePlugin: DataStoreCategoryPlugin {
     /// them to `StorageEngine.setUp(modelSchemas:)`
     public func configure(using amplifyConfiguration: Any?) throws {
         modelRegistration.registerModels(registry: ModelRegistry.self)
+        for modelSchema in ModelRegistry.modelSchemas {
```

Comment:
do we need to reset the `dispatchedModelSyncedEvents` map on `configure()` and `reset()` ?

Reply:
`reset` is used internally for testing, so it looks okay that we do not set `dispatchedModelSyncedEvents = [:]` since configure will just add new ones/replace existing entries. when the model registry changes, then `dispatchedModelSyncedEvents` may have extra entries if this happens, which is just held in memory won't get used by the plugin. so, it make sense to set it back to `[:]`  on reset